### PR TITLE
[LDAP] Free the search result after a search to free memory

### PR DESCRIPTION
**Pull request by hiddewie**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by fabpot** [comment]
ping @csarrazi

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 1]
```diff
@@ -86,6 +86,8 @@ public function find($dn, $query, $filter = '*')
         $search = ldap_search($this->connection, $dn, $query, $filter);
         $infos = ldap_get_entries($this->connection, $search);
 
+        ldap_free_result($search);
```
Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: `ldap_free_results` won't work if the `ldap_search` method call returns `false` or throws an error.

You should add some error handling code around the statement.

**Comment by hiddewie** [comment]
I've updated the code to check for errors during the search of free.

**Comment by csarrazi** [comment]
:+1:

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 2]
```diff
@@ -84,8 +84,15 @@ public function find($dn, $query, $filter = '*')
         }
 
         $search = ldap_search($this->connection, $dn, $query, $filter);
+        if (false === $search) {
```
Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: Add newline before the `if` statement.

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 3]
```diff
@@ -84,8 +84,15 @@ public function find($dn, $query, $filter = '*')
...
+            throw new LdapException(ldap_error($this->connection));
+        }
```
Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: Same thing after the bracket.

**Comment by csarrazi** [comment]
LGTM. Just 2 comments regarding style, so it is consistent with the rest.

**Comment by hiddewie** [comment]
I've updated the spacing.

**Comment by csarrazi** [comment]
By the way, could you add a few functional tests?

**Comment by hiddewie** [comment]
I've added a `find` test and a few for checking the (new) exceptions.

**Comment by csarrazi** [comment]
Nevermind my comment.

I thought you were in the `master` branch. Could you revert the last commit, then squash your all your commits in a single one? Thanks!

While we could add functional tests, these will most likely break when merging back to the `master` branch.

**Comment by hiddewie** [comment]
Yes, reverted last commits.

**Comment by csarrazi** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @hiddewie.

**Pull request merged**